store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -38,10 +39,39 @@ public class OwaspHtmlSanitizer implements Callable<String> {
     private String vHost;
 
     public OwaspHtmlSanitizer(String html, boolean neuterImages, String vHost) {
-        this.html = html;
+        // fix to check extra tags & remove
+        this.html = checkUnbalancedTags(html);
         this.neuterImages = neuterImages;
         this.vHost = vHost;
     }
+    
+    /** 
+     * A method to check & remove unbalanced tags which may have started, not ended
+     * 
+     * @return the Formatted Html after removing unbalanced tags
+     */
+    private String checkUnbalancedTags(String html) {
+        ArrayList<String> tags = new ArrayList<String>();
+        // in tags array we can add multiple tags, so for each tag write start &
+       // end with a & separator like(<!--&-->)
+        tags.add("<!--&-->");
+        String formattedHtml = "";
+        for (String str : tags) {
+            String start = str.split("&")[0];
```

Comment:
Please do operation str.split once and get index 0 and 1  separately

Reply:
done with the changes